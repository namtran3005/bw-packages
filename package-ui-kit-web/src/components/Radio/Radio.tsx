import * as React from 'react';
import styled from 'styled-components';
import { Radio as MuiRadio } from '@material-ui/core';
import { RadioProps as MuiRadioProps } from '@material-ui/core/Radio';
import { Colors, ColorsType } from '@bitwala-cryptobank-squad/package-theme';

interface RadioProps extends MuiRadioProps {
  foreground?: ColorsType;
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const RootComponent: React.FunctionComponent<RadioProps> = (props) => (
  <MuiRadio {...props} />
);

const defaultForeground: ColorsType = 'primaryLilac';
const defaultDisabledForeground: ColorsType = 'primaryGrey';

const Root = styled(RootComponent)`
  &&& {
    color: ${(props) =>
      Colors.get(
        !props.disabled
          ? props.foreground || defaultForeground
          : defaultDisabledForeground
      )};
    &:hover {
      background-color: transparent;
      &:not(.checked) {
        .MuiSvgIcon-root {
          box-shadow: 0 0 0 6px transparent,
            inset 0 0 0 4px
              ${(props) =>
                Colors.getWithAlpha(
                  props.foreground || defaultForeground,
                  0.25
                )};
        }
      }
    }

    .MuiSvgIcon-root {
      border-radius: 50%;
      transition: all 0.4s;
    }
  }
`;

Root.displayName = 'Radio';
RootComponent.displayName = 'RadioComponent';

const Radio: React.FunctionComponent<RadioProps> = (props) => (
  <Root
    disableFocusRipple
    disableRipple
    disableTouchRipple
    classes={{
      checked: 'checked',
    }}
    {...props}
  />
);

export { Radio, RadioProps };
