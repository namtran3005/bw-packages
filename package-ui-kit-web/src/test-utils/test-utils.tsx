import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
