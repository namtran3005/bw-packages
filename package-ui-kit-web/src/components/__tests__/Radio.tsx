import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { Radio } from '../Radio';

const defaultProps = {};

describe('Radio', () => {
  it('should render radio', () => {
    const { baseElement } = renderWithTestHarness(<Radio {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });
});
