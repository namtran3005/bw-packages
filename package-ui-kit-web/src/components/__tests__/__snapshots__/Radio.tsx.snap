// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render radio 1`] = `
.c0.c0.c0 {
  color: #BEAAFF;
}

.c0.c0.c0:hover {
  background-color: transparent;
}

.c0.c0.c0:hover:not(.checked) .MuiSvgIcon-root {
  box-shadow: 0 0 0 6px transparent, inset 0 0 0 4px rgba(190,170,255,0.25);
}

.c0.c0.c0 .MuiSvgIcon-root {
  border-radius: 50%;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

<body>
  <div>
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary c0 MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-4"
          type="radio"
          value=""
        />
        <div
          class="PrivateRadioButtonIcon-root-5"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </div>
      </span>
    </span>
  </div>
</body>
`;
